### DISCUSSION TITLE
Invoke a function to add two numbers

### DISCUSSION THREAD

**Discussion by agusglez** [gudzpoz/luajava#197] [Q&A]
Hi,

I would like to write some code that will invoke below LUA function passing two numbers and get the result.

function MyAdd( num1, num2 )
    return num1 + num2
end

Can anyone help me with that? From the documentation I don't get how to do it.

**Comment by gudzpoz** [comment]
Lua function invocation from Java code is explained in [Java API#Call Lua functions](https://gudzpoz.github.io/luajava/java.html#call-lua-functions). Following that example, you will need to:

```java
try (Lua L = new Lua54()) {
    L.run("function MyAdd(num1, num2) return num1 + num2 end");
    LuaValue add = L.eval("return MyAdd")[0];
    LuaValue i = add.call(1, 2)[0];
    assertEquals(3, i.toInteger());
}
```

I guess the documentation is not clear enough? Would you mind pointing out what aspects of the documentation confuses you so that I can improve it? Thanks!

**Reply by agusglez** [reply]
I missed an example like this one, where we create our function (MyAdd) and we invoke it.
Btw, how can I do the same but with an anonymous function?
    L.run("function(num1, num2) return num1 + num2 end");
Thanks for your help @gudzpoz

**Reply by agusglez** [reply]
@gudzpoz Would you be able to answer my question?
